test/module/aa-module.specs.ts

```diff
@@ -0,0 +1,1057 @@
+      );
+
+      const nonceFromContract2 = await verifyingSingletonPaymaster[
+        "getSenderPaymasterNonce(address)"
+      ](expectedSmartAccountAddress);
+
+      const hash2 = await verifyingSingletonPaymaster.getHash(
+        userOp2,
+        nonceFromContract2.toNumber(),
+        await offchainSigner.getAddress()
+      );
+      const sig2 = await offchainSigner.signMessage(arrayify(hash2));
+      const userOpAA2 = await fillAndSign(
+        {
+          ...userOp2,
+          paymasterAndData: hexConcat([
+            paymasterAddress,
+            ethers.utils.defaultAbiCoder.encode(
+              ["address", "bytes"],
+              [await offchainSigner.getAddress(), sig2]
+            ),
+          ]),
+        },
+        walletOwner,
+        entryPoint
+      );
+      console.log(userOpAA2);
+      await entryPoint.handleOps(
+        [userOpAA2],
+        await offchainSigner.getAddress(),
+        {
+          gasLimit: 10000000,
+        }
+      );
+    });
+
+    it("send transaction from enabled social recovery module using EntryPoint (and Paymaster)", async function () {
+      // Now here also the wallet with owner and index 2 should have been deployed
+
+      const expectedSmartAccountAddress =
+        await walletFactory.getAddressForCounterfactualAccount(owner, 2);
+
+      userSCW = await ethers.getContractAt(
+        "contracts/smart-contract-wallet/SmartAccount.sol:SmartAccount",
+        expectedSmartAccountAddress
+      );
+
+      const prevNonce = await userSCW.nonce();
+      console.log("previous nonce is: ", prevNonce.toNumber());
+
+      await token
+        .connect(accounts[0])
+        .transfer(expectedSmartAccountAddress, ethers.utils.parseEther("100"));
+
+      // module is already deployed, enabled and destination whitelisted
+
+      const isEnabled = await userSCW.isModuleEnabled(
+        socialRecoveryModule.address
+      );
+      expect(isEnabled).to.be.equal(true);
+
+      /* expect(
+        await userSCW.getModulesPaginated(AddressOne, 10)
+      ).to.be.deep.equal([[socialRecoveryModule.address], AddressOne]); */
+
+      // invoking transaction via enabled module
+
+      const SmartAccount = await ethers.getContractFactory("SmartAccount");
+      const SocialRecoveryModule = await ethers.getContractFactory(
+        "SocialRecoveryModule"
+      );
+
+      const newOwner = accounts[5];
+
+      // bob confirms transaction for setOwner()
+      tx = await socialRecoveryModule
+        .connect(accounts[1])
+        .confirmTransaction(userSCW.address, newOwner.address);
+      console.log(await userSCW.owner());
+
+      // charlie confirms transaction for setOwner()
+      tx = await socialRecoveryModule
+        .connect(accounts[2])
+        .confirmTransaction(userSCW.address, newOwner.address);
+
+      // have to make entrypoint do confirmTransaction somehow
+
+      const txnDataModule = SocialRecoveryModule.interface.encodeFunctionData(
+        "recoverAccess",
```

Comment:
let's say we want to use a function, that is protected by Module own access contraol instead of recoverAccess?
When this access control can take the signature to verify from?

Reply:
yes, recoverAccess was actually access controlled, i commented out the modifier for test case ( otherwise its tricky like making wallet address / entrypoint as recoverer) 
ideal is being able to pass signature for access controlled module methods, so this flow can work
EP -> Wallet -> Module -> Wallet

Reply:
Yep, I think we should pass it, and it even can become a standard for aa-flow compatible modules.
Ideally it should be the same signature from userOp

Reply:
"Ideally it should be the same signature from userOp" what do you mean?
recoverer to sign over userOp?
I was thinking it's usually defined typehash within module contract (say recovery data struct and EIP712 typehash) as the module contract itself verifies that sig for the action being done for passed data.